cmd/run.go

```diff
@@ -2,106 +2,49 @@ package cmd
 
 import (
 	"fmt"
-	"log"
-	"os"
 
 	"github.com/odpf/meteor/agent"
-	"github.com/odpf/meteor/config"
-	"github.com/odpf/meteor/internal/logger"
-	"github.com/odpf/meteor/internal/metrics"
-	"github.com/odpf/meteor/plugins"
+	"github.com/odpf/meteor/metrics"
 	"github.com/odpf/meteor/recipe"
 	"github.com/odpf/meteor/registry"
+	"github.com/odpf/salt/log"
 	"github.com/spf13/cobra"
 )
 
-// RunCmd creates a command object for the "run" action
-func RunCmd() *cobra.Command {
-	cmd := &cobra.Command{
+// RunCmd creates a command object for the "run" action.
+func RunCmd(lg log.Logger, mt *metrics.StatsdMonitor) *cobra.Command {
+	return &cobra.Command{
 		Use:     "run [COMMAND]",
-		Short:   "Run meteor for provided receipes",
 		Example: "meteor run recipe.yaml",
+		Short:   "Run meteor for provided recipes.",
 		Args:    cobra.ExactArgs(1),
-		Run: func(cmd *cobra.Command, args []string) {
-			fi, err := os.Stat(args[0])
-			if err != nil {
-				fmt.Println(err)
-				return
-			}
-			switch mode := fi.Mode(); {
-			case mode.IsDir():
-				rundir(args[0])
-			case mode.IsRegular():
-				run(args[0])
-			}
-		},
-	}
-
-	return cmd
-}
+		RunE: func(cmd *cobra.Command, args []string) error {
 
-// Run a single recipe
-func run(recipeFile string) {
-	c, err := config.LoadConfig()
-	if err != nil {
-		log.Fatal(err)
-	}
-	log := logger.New(c.LogLevel)
-	plugins.Log.Level = log.Level
-
-	reader := recipe.NewReader()
-	rcp, err := reader.Read(recipeFile)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	run := initRunner(c, log).Run(rcp)
-	if run.Error != nil {
-		log.Fatal(run.Error)
-	}
-	log.WithField("run", run).Info("Done!")
-}
+			runner := agent.NewAgent(registry.Extractors, registry.Processors, registry.Sinks, mt)
 
-// Run a directory of recipes
-func rundir(dirPath string) {
-	c, err := config.LoadConfig()
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	reader := recipe.NewReader()
-	recipeList, err := reader.ReadDir(dirPath)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	log := logger.New(c.LogLevel)
-	plugins.Log.Level = log.Level
+			reader := recipe.NewReader()
+			recipes, err := reader.Read(args[0])
+			if err != nil {
+				return err
+			}
 
-	runs := initRunner(c, log).RunMultiple(recipeList)
-	log.WithField("runs", runs).Info("Done!")
-}
+			count := len(recipes)
```

Comment:
few rules I personally follow to decide if we should create a new variable in these cases
1. Is this value getting used in more than one place? If no, don't create
2. Is creating a variable out of it helps create an alias and them improves code readability? If no, and its already very obvious to read, don't create
3. Is creating a new variable cache a compute-intensive operation result? If no, don't create

So for example:
```
reader := recipe.NewReader()
recipes, err := reader.Read(args[0])
if err != nil {
    return err
}
```
can be written as
```
recipes, err := recipe.NewReader().Read(args[0])
if err != nil {
    return err
}
```
and 
```
if len(recipes) == 0 {
    //..
}
```

Reply:
Agree. I was planning to use count for printing more things like how many recipes ran etc. Moved that scope to next PR.